digdag-standards/src/main/java/io/digdag/standards/operator/redshift/BaseRedshiftLoadOperator.java

```diff
@@ -0,0 +1,230 @@
+    @Override
+    protected RedshiftConnectionConfig configure(SecretProvider secrets, Config params)
+    {
+        return RedshiftConnectionConfig.configure(secrets, params);
+    }
+
+    /* TODO: This method should be in XxxxConnectionConfig ? */
+    @Override
+    protected RedshiftConnection connect(RedshiftConnectionConfig connectionConfig)
+    {
+        return RedshiftConnection.open(connectionConfig);
+    }
+
+    @Override
+    protected SecretProvider getSecretsForConnectionConfig()
+    {
+        return context.getSecrets().getSecrets("aws.redshift");
+    }
+
+    protected abstract List<SecretProvider> additionalSecretProvidersForCredentials(SecretProvider awsSecrets);
+
+    protected AWSCredentials createBaseCredential(SecretProvider secretProvider)
+    {
+        SecretProvider awsSecrets = secretProvider.getSecrets("aws");
+        SecretProvider redshiftSecrets = awsSecrets.getSecrets("redshift");
+        List<SecretProvider> secretProviders = additionalSecretProvidersForCredentials(awsSecrets);
+
+        String keyOfAccess = "access-key-id";
+        String accessKeyId =
+                secretProviders.stream()
+                        .map(sp -> sp.getSecretOptional(keyOfAccess))
+                        .filter(Optional::isPresent)
+                        .findFirst()
+                        .orElse(Optional.absent())
+                        .or(redshiftSecrets.getSecretOptional(keyOfAccess))
+                        .or(() -> awsSecrets.getSecret(keyOfAccess));
+
+        String keyOfSecret = "secret-access-key";
+        String secretAccessKey =
+                secretProviders.stream()
+                        .map(sp -> sp.getSecretOptional(keyOfSecret))
+                        .filter(Optional::isPresent)
+                        .findFirst()
+                        .orElse(Optional.absent())
+                        .or(redshiftSecrets.getSecretOptional(keyOfSecret))
+                        .or(() -> awsSecrets.getSecret(keyOfSecret));
+
+        return new BasicAWSCredentials(accessKeyId, secretAccessKey);
+    }
+
+    private AWSSessionCredentials createSessionCredentials(Config config, AWSCredentials baseCredential)
+    {
+        List<AcceptableUri> acceptableUris = buildAcceptableUriForSessionCredentials(config, baseCredential);
+
+        if (!config.get("temp_credentials", Boolean.class, true)) {
+            return new BasicSessionCredentials(
+                    baseCredential.getAWSAccessKeyId(),
+                    baseCredential.getAWSSecretKey(),
+                    null
+            );
+        }
+
+        AWSSessionCredentialsFactory sessionCredentialsFactory =
+                new AWSSessionCredentialsFactory(
+                        baseCredential.getAWSAccessKeyId(),
+                        baseCredential.getAWSSecretKey(),
+                        acceptableUris);
+
+        Optional<String> roleArn = config.getOptional("role_arn", String.class);
```

Comment:
EMR operator implements that `role_arn` and `role_session_name` are secret-only parameters. To be consistent with it, here should get `role_arn` and `role_session_name` from SecretProvider(s).